src/v3/bandwidthRtc.ts

```diff
@@ -346,29 +378,88 @@ export class BandwidthRtc {
     });
   }
 
-  private setupPublishingPeerConnection(): RTCPeerConnection {
+  private async addHeartbeatDataChannel(peerConnection: RTCPeerConnection): Promise<RTCDataChannel> {
```

Comment:
Any reason this is async/returns a promise? I don't see anything in these functions that would necessitate that.